Dockerfile

```diff
@@ -13,16 +16,34 @@ RUN apt-get update && \
     rm -rf /var/lib/apt/lists/*
 
 # Copy the current directory contents into the container at /app
-COPY . /app
+COPY requirements.txt .
+
+# Make a virtualenv that we can copy across multistage builds
+ENV VIRTUAL_ENV=/venv
+RUN python -m venv $VIRTUAL_ENV
+# "Activate" the virtualenv
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
```

Comment:
Why not use a system environment with `pip wheel`?

Reply:
you could do a pip install and then copy the packages over from the lib directory like chainguard does https://edu.chainguard.dev/chainguard/chainguard-images/reference/python/

Reply:
A virtualenv is not necessary in this case. We'll never have multiple environments in this docker container. A system-wide install is better and results in less unneccesary files (due to the virtualenv).

The following sample is a very explicit and minimal way to install from wheel-directory that minimizes container size.

```
COPY --from=python-build-stage /usr/src/app/wheels  /wheels/

RUN pip install --no-cache-dir --no-index --find-links=/wheels/ /wheels/* \
  && rm -rf /wheels/
```